mantidproject/mantid
# Engineering GUI hook up GSAS tab to GSASIIRefineFitPeaks

The GSAS tab of the ED GUI was hooked up to the algorithm GSASIIRefineFitPeaks, allowing Rietveld refinement on multiple runs from the GUI.

**To test:**

You'll need GSAS-II installed. Install it using one of the scripts in `scripts/GSAS-II` or follow the instructions [here](https://subversion.xray.aps.anl.gov/trac/pyGSAS). Test data is in [GSASIIRefineFitPeaks.zip](https://github.com/mantidproject/mantid/files/1739753/GSASIIRefineFitPeaks.zip)

Start the ED GUI by going to `Interfaces -> Diffraction -> Engineering Diffraction`. You'll be prompted to enter an RB number if you haven't opened the GUI before - do so then go to the `GSAS-II Refinement` tab.

Load the run by clicking the first **Browse** button, selecting `ENGINX_280625_focused_bank_1.nxs` from the zip, and clicking **Load**. `280625_1` should appear in the list in the bottom right of the GUI - click this and the run should appear in the plot.

Select `template_ENGINX_241391_236516_North_bank.prm` for the instrument parameter file, all three `.cif` files for the phase files, a new location to save the gsas project (`.gpx`) to, and your GSAS-II installation directory (containing `GSASIIscriptable.py`), all with their respective browse buttons.

Enter 10000 for **XMin**, 40000 for **XMax**, and select to refine sigma and gamma. Click **Do Refinement**. The fitted peaks should appear in the plot, and the fit output values should appear in the **Fit Results** box. They should be:

![capture](https://user-images.githubusercontent.com/29916298/36488975-bd9df582-171c-11e8-8368-20d753da2f2c.PNG)

Try zooming in and out on the plot, removing the run, plotting, and generally just fiddle with the parameters and see if you can break anything. In particular, make sure that putting garbage into any of the inputs is handled nicely.

Fixes #15552 

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/pull/21804/commits/feed6005bcffbda4fa3bdbd346e06ffcac1cb463)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
